File: android/src/main/kotlin/jyahann/dgis_map_kit/DGisMapConfig.kt
import jyahann.dgis_map_kit.CameraUtils
import ru.dgis.sdk.map.CameraPosition

class DGisMapConfig(creationParams: Map<String?, Any?>?) {
    var token: String
    var initialCameraPosition: CameraPosition
    var layers: List<Map<String, Any?>>
    var theme: String
    var enableMyLocation: Boolean

    init {
        this.token = creationParams?.get("token") as String
        this.enableMyLocation = creationParams?.get("enableMyLocation") as Boolean
        this.initialCameraPosition = CameraUtils.getCameraPositionFromDart(creationParams?.get("initialCameraPosition") as Map<String, Any>)
        this.theme = creationParams?.get("theme") as String
        this.layers = (creationParams?.get("layers") as List<Any>).map { layer -> layer as Map<String, Any?> }
    }
}

File: ios/Classes/DGisMapConfig.swift
import DGis

class DGisMapConfig {
     var keyFile: String;
     var initialCameraPosition: CameraPosition;
     var layers: Array<Dictionary<String, Any?>>;
     var theme: String;

     init (
         arguments args: Dictionary<String?, Any?>
     ) {
         self.keyFile = args["keyFile"] as! String;
         self.initialCameraPosition = CameraUtils.getCameraPositionFromDart(
            cameraPosition: args["initialCameraPosition"] as! Dictionary<String, Any>
         );
         self.layers = (args["layers"] as! Array<Any>).map{$0 as! Dictionary<String, Any?>}
         self.theme = args["theme"] as! String;
     }
 }
